### ISSUE TITLE
What's the best way to test changes on an extension?

### ISSUE THREAD

**Issue by leinardi** [Ulauncher/Ulauncher#402]
Sorry for the silly question but I was not able to find an explanation on the Docs: I have already published 2 extensions and I have recently updated them to support the API v2. 
Now I would like to make use of the new feature available on the new API but I don't know how to test my changes without pushing them to master on GitHub.

Is there a way to run locally an extension? Or is there a way to pick a specific branch from GitHub?

I tried to copy the extension source on `~/.cache/ulauncher_cache/extensions` but it doesn't seem to work:
![image](https://user-images.githubusercontent.com/273338/59964393-3435f200-9500-11e9-9ba8-cce4b02b0e46.png)

If I add the same extension code from GitHub it works properly:
![image](https://user-images.githubusercontent.com/273338/59964405-73fcd980-9500-11e9-9531-c8569f9bb450.png)

In short I would like to know how can I test changes on an already published extension without having to push my changes to production.

**Comment by friday** [comment]
I've basically been doing this while adding v5 support:

1. Add manifest changes and push to github blindly without python 3 migration.
2. Install extension through github.
3. Watch stack traces in terminal. And fix, then copy the file directly and restart ulauncher to see the next error.

This worked for me:
```sh
pkill ulauncher
cp *.py ~/.cache/ulauncher_cache/extensions/com.github.me.my-extension
ulauncher -v
```

**Comment by cryobry** [comment]
Install the extension locally, see: https://github.com/Ulauncher/Ulauncher/issues/388#issuecomment-497143959

**Comment by gornostal** [comment]
> I don't know how to test my changes without pushing them to master on GitHub

The simplest way is to create create a new dir under `~/.cache/ulauncher_cache/extensions` and then use Ulauncher with `--no-extensions` command.

More details here: http://docs.ulauncher.io/en/latest/extensions/debugging.html#debugging-logging
